include/alpaka/rand/Philox/PhiloxSingle.hpp

```diff
@@ -0,0 +1,162 @@
+                 */
+                ALPAKA_FN_HOST_ACC void advanceState()
+                {
+                    this->advanceCounter(state.counter);
+                    state.result = this->nRounds(state.counter, state.key);
+                    state.position = 0;
+                }
+
+                /** Get the next random number and advance internal state
+                 *
+                 * The intermediate result stores N = TParams::counterSize numbers. Check if we've already given out
+                 * all of them. If so, generate a new intermediate result (this also resets the pointer to the position
+                 * of the actual number). Finally, we return the actual number.
+                 *
+                 * @return The next random number
+                 */
+                ALPAKA_FN_HOST_ACC auto nextNumber()
+                {
+                    state.position++;
+                    if(state.position == TParams::counterSize)
+                    {
+                        advanceState();
+                    }
+#ifdef PLAIN_GPU_VECTOR_TYPES
+                    switch(state.position)
+                    {
+                    case 0:
+                        return state.result.x;
+                    case 1:
+                        return state.result.y;
+                    case 2:
+                        return state.result.z;
+                    case 3:
+                        return state.result.w;
+                    default: // Should never get here
+                        return 0u;
+                    }
+#else
+                    return state.result[state.position];
+#endif
+                }
+
+                /// Skips the next \a offset numbers
+                ALPAKA_FN_HOST_ACC void skip(uint64_t offset)
+                {
+                    state.position += offset & 3;
+                    offset += state.position < 4 ? 0 : 4;
+                    state.position += state.position < 4 ? 0 : -4;
+                    this->skip4(offset / 4);
+                }
+
+            public:
+                /** Construct a new Philox engine with single-value output
+                 *
+                 * @param seed Set the Philox generator key
+                 * @param subsequence Select a subsequence of size 2^64
+                 * @param offset Skip \a offset numbers form the start of the subsequence
+                 */
+                ALPAKA_FN_HOST_ACC PhiloxSingle(uint64_t seed = 0, uint64_t subsequence = 0, uint64_t offset = 0)
+                    : state{
+                        {0, 0, 0, 0},
+                        {static_cast<uint32_t>(lo32(seed)), static_cast<uint32_t>(hi32(seed))},
+                        {0, 0, 0, 0},
+                        0}
+                {
+                    this->skipSubsequence(subsequence);
+                    skip(offset);
+                    advanceState();
+                }
```

Comment:
I think I would have written something like this:
```suggestion
                    : state{} // bulk zero initialize
                {
                    state.seedLow = lo32(seed); // however seedLow is called
                    state.seedHigh = hi32(seed);
                    this->skipSubsequence(subsequence);
                    skip(offset);
                    advanceState();
                }
```
I would be too afraid that I might miss initializing a value in `state` and have a bug.

Reply:
I don't really see the problem here - you could forget to initialize the seed as is in the second example anyway.